source/common/router/router.cc

```diff
@@ -503,20 +503,39 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   const auto& upstream_http_protocol_options = cluster_->upstreamHttpProtocolOptions();
 
   if (upstream_http_protocol_options.has_value()) {
-    const auto parsed_authority = Http::Utility::parseAuthority(headers.getHostValue());
-    if (!parsed_authority.is_ip_address_ && upstream_http_protocol_options.value().auto_sni()) {
-      callbacks_->streamInfo().filterState()->setData(
-          Network::UpstreamServerName::key(),
-          std::make_unique<Network::UpstreamServerName>(parsed_authority.host_),
-          StreamInfo::FilterState::StateType::Mutable);
+    std::string sni_value;
```

Comment:
```suggestion
    absl::string_view sni_value;
```
keep this string_view to the end

Reply:
Thanks for the suggestion!